specification/liftrastronomer/Astronomer.Astro.Management/LiftrBase/main.tsp

```diff
@@ -65,6 +69,10 @@ model OfferDetails {
 
   @doc("Plan Display Name for the marketplace offer")
   termId?: string;
+
+  @added(Versions.v2024_08_27_preview)
+  @doc("Subscription auto renewal flag")
+  autoRenew?: boolean;
```

Comment:
>autoRenew [](http://example.com/codeflow?start=2&length=9)

Standard guidance is: replace boolean/switch properties with a more meaningful enum whenever possible. 

A boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated]. 

Note: do NOT define a 'boolean enum' with two values 'True and False'. This might be easier to 'extend' in terms of types, but semantically its confusing, and no better than a boolean.

Reply:
As per the requirements from PM, default value of the autoRenew is true. So, we have only two possible values, and I am not seeing any use case where it can have third value. Either autoRenew is enabled or disabled (true or false).

Reply:
Also, we need to pass this property to marketplace as it is (Marketplace team is expecting true or false for this property).

Reply:
You could still accept enums and send the boolean to Saas RP if they need it so. 

We have seen cases where the boolean values are usually not enough as years go by and more values are possible for this state. 
This is a best practice guidance which helps your API to be future proof. This would be good time to make the change, else boolean will keep going on in the next API versions too. 

Please give another thought on this.

Reply:
Defined enum for auto-renew.

Reply:
This is essentially a `boolean enum` (Enable/Disable vs True/False) as mentioned above. How about taking @ramoka178 's suggestion to change it to something like `renewalMode`. It's clearer and more expandable.

> Note: do NOT define a 'boolean enum' with two values 'True and False'. This might be easier to 'extend' in terms of types, but semantically its confusing, and no better than a boolean.

Reply:
@gary-x-li, are you taking about renaming AutoRenewStatus to RenewalMode or something else?

Reply:
@gary-x-li, Updated code as per suggestion.